### ISSUE TITLE
Please add support for shopspring/decimal

### ISSUE THREAD

**Issue by jmcalves275** [swaggo/swag#112]
I have an error when I run `swag init`.

product.go
```
type(
        
        Product struct {
		ID        int      `json:"id"`
		Offers    []*Offer `json:"offers"`
	}

	Offer struct {
		ID           int             `json:"id"`
		SKU          string          `json:"sku"`
		Price        decimal.Decimal `json:"price"`
		PriceWithVAT decimal.Decimal `json:"price_with_vat"`
		Currency     string          `json:"currency"`
		Stock        int             `json:"stock"`
		Merchant     *Merchant       `json:"merchant"`
	}

)	
```

domain.go
```
// API List products godoc
// @Tags products
// @Summary List products
// @Produce  json
// @Success 201 {object} products.Product ""
// @Failure 400 {object} errors.ErrorResponse
// @Failure 500 {object} errors.ErrorResponse
// @Router /products/ [get]
```

**Error:**
```
Decimal is not supported. but it will be set with string temporary. 
Please report any problems.Decimal is not supported. 
but it will be set with string temporary. 
Please report any problems.2018/05/08 11:31:42 Generating products.Merchant
panic: not supported&{773 0xc4204fd830}
```

As you notice, I am having problems with type Decimal, but also with pointer (to Merchant) inside Offer struct. Can you also add support for pointers?

**Comment by pei0804** [comment]
OK. I check.
And I add plugin for special types.

**Comment by pei0804** [comment]
I support ptr slice and Decimal.
I don't understand problem in products.Merchant.
How do you have struct. Please more info.

**Comment by pei0804** [comment]
Probably all was solved.
Please wait for the merge.

**Comment by easonlin404** [comment]
@jmcalves275  Fix via #113. please update swag to  the latest version and reopen this issue if have any problems.
And we will create plugin function asap to support for any specific type that is not basic types. 🎉 🎉
